packages/kujua-sms/kujua-sms/updates.js

```diff
@@ -356,19 +356,20 @@ exports.update_message_task = function(doc, request) {
  * Return task object that matches message uuid or a falsey value if match
  * fails.
  */
-var getTask = function(uuid, doc, type) {
-    type = type || 'message';
-    if (!uuid || !doc || !doc.tasks) {
-        return;
-    }
-    if (type === 'message') {
-        for (var i in doc.tasks) {
-            for (var j in doc.tasks[i].messages) {
-                if (uuid === doc.tasks[i].messages[j].uuid) {
-                    return doc.tasks[i];
-                }
+var getTask = function(uuid, doc) {
+    for (var i in doc.tasks) {
+        for (var j in doc.tasks[i].messages) {
```

Comment:
Yes, and yes. `jshint` would complain because it has no way of knowing they're arrays.